blockchain/utxobackend.go

```diff
@@ -174,40 +274,32 @@ func removeRegressionDB(net wire.CurrencyNet, dbPath string) error {
 // LoadUtxoDB loads (or creates when needed) the UTXO database and returns a
 // handle to it.  It also contains additional logic such as ensuring the
 // regression test database is clean when in regression test mode.
-func LoadUtxoDB(params *chaincfg.Params, dataDir string) (database.DB, error) {
-	// Set the database path based on the data directory and database name.
-	dbPath := filepath.Join(dataDir, utxoDbName)
+func LoadUtxoDB(params *chaincfg.Params, dataDir string) (*leveldb.DB, error) {
+	// Set the database path based on the data directory, UTXO database directory,
+	// and database name.
+	dbPath := filepath.Join(dataDir, utxoDbDir, utxoLdbName)
 
 	// The regression test is special in that it needs a clean database for each
 	// run, so remove it now if it already exists.
 	removeRegressionDB(params.Net, dbPath)
 
-	// createDB is a convenience func that creates the database with the type and
-	// network specified in the config at the path determined above while also
-	// creating any intermediate directories in the configured data directory path
-	// as needed.
-	createDB := func() (database.DB, error) {
-		// Create the data dir if it does not exist.
-		err := os.MkdirAll(dataDir, 0700)
-		if err != nil {
-			return nil, err
-		}
-		return database.Create(utxoDbDefaultDriver, dbPath, params.Net)
+	// Create the data dir if it does not exist.
+	err := os.MkdirAll(dataDir, 0700)
```

Comment:
I updated this to only attempt to create the dirs if the database does not exist.  I also added an additional comment regarding the behavior of `os.MkdirAll` seen on some less supported OSes that you brought up:
```
// NOTE: It is important that os.MkdirAll is only called if the database
// does not exist.  The documentation states that os.MidirAll does nothing
// if the directory already exists.  However, this has proven not to be the
// case on some less supported OSes and can lead to creating new directories
// with the wrong permissions or otherwise lead to hard to diagnose issues.
```